server/core/storelimit/sliding_window.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storelimit
+
+import (
+	"github.com/tikv/pd/pkg/utils/syncutil"
+)
+
+const (
+	// minSnapSize is the min value to check the windows has enough size.
+	minSnapSize = 10
+)
+
+var _ StoreLimit = &SlidingWindows{}
+
+// SlidingWindows is a multi sliding windows
+type SlidingWindows struct {
+	mu      syncutil.RWMutex
+	windows []*window
+}
+
+// NewSlidingWindows is the construct of SlidingWindows.
+func NewSlidingWindows(capacity int64) *SlidingWindows {
```

Comment:
I think we can use float64 to align with Reset

Reply:
fixed